File: code/python/method_init_in_runtime.py
class ProccessElements:
  def __init__(self):
    def element(el_name):
      def render(content):
        return '<{0}>{1}</{0}>'.format(el_name, content)
      return render

    for el_name in self.elements:
      setattr(self, el_name, element(el_name))


class HtmlELements(ProccessElements):
  elements = ('div', 'span')

print(HtmlELements().div('hello'))

File: code/ruby/method_init_in_runtime.rb
class ProccessElements
  def self.element(el_name)
    define_method "#{el_name}_element" do |content|
      "<#{el_name}>#{content}</#{el_name}>"
    end
  end
end

class HtmlELements < ProccessElements
  element :div
  element :span
end

puts HtmlELements.new.div_element('hello')
